packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -561,3 +568,38 @@ export class PerformanceEvent {
         throw error;
     }
 }
+
+/**
+     * Take in a event object, stringify any fields that are non-primitives, and return the new event object.
+     * @param event - Event with fields you want to stringify.
+     */
+function stringifyEventFields(event: TelemetryEventTypes): void {
+    for (const key of Object.keys(event)) {
+        const filteredEventVal = filterValidTelemetryProps(event[key]);
+        if (filteredEventVal !== null) {
+            event[key] = filteredEventVal;
+        }
+    }
+}
+/**
+ * Takes in parameter, if parameter is of primitive type, return the original value.
+ * If parameter is an array, stringify then return the result.
+ * @param x - parameter passed to validate/filter
+ */
+function filterValidTelemetryProps(x: any): TelemetryEventPropertyType | null {
```

Comment:
Either import this function from `errorLogging.ts` (or move it somewhere else in this package and use it in both places), or define your own function here, something like

`convertToBasePropertyType(x: TelemetryEventPropertyType): TelemetryBasePropertyType`

I'm leaning towards the separate `convertToBaseProperty` function, since I think `filterValidTelemetryProps` and related code in errorLogging.ts can go away once all this work is done, right?